afl build broken

Looks like https://github.com/stellar/stellar-core/pull/1617 broke the AFL build.

Unclear if this is because of unsupported options or what.
Building https://github.com/stellar/docker-stellar-core/blob/master/Dockerfile.fuzz
yields:
```
afl-clang-fast++ -DHAVE_CONFIG_H -I. -I..  -DSQLITE_OMIT_LOAD_EXTENSION=1 -isystem ".." -I"../src" -I"../src" -isystem /stellar-core/lib/libsodium/src/libsodium/include -isystem ../lib/libsodium/src/libsodium/include -isystem ../lib/libsodium/src/libsodium/include/sodium -isystem /stellar-core/lib/xdrpp -isystem /stellar-core/lib/xdrpp -isystem ../lib/libmedida/src -isystem ../lib/soci/src/core -isystem ../lib/soci/src/backends/sqlite3 -isystem ../lib/soci/src/backends/postgresql  -DASIO_SEPARATE_COMPILATION=1 -DASIO_STANDALONE -isystem ../lib/asio/asio/include -isystem "../lib" -isystem "../lib/autocheck/include" -isystem "../lib/cereal/include" -isystem "../lib/util" -isystem "../lib/soci/src/core" -DUSE_POSTGRES=1 -I/usr/include/postgresql    -g -O2 -std=c++11  -pthread -Wall -Wno-unused-command-line-argument -Qunused-arguments -Wno-unused-local-typedef -Wno-unknown-warning-option -Werror=unused-result -MT soci/src/backends/sqlite3/standard-into-type.o -MD -MP -MF $depbase.Tpo -c -o soci/src/backends/sqlite3/standard-into-type.o soci/src/backends/sqlite3/standard-into-type.cpp &&\
mv -f $depbase.Tpo $depbase.Po
In file included from soci/src/backends/sqlite3/standard-into-type.cpp:9:
../lib/soci/src/backends/sqlite3/soci-sqlite3.h:45:10: fatal error: 'sqlite3.h' file not found
#include <sqlite3.h>
         ^
1 error generated.
Makefile:1701: recipe for target 'soci/src/backends/sqlite3/standard-into-type.o' failed
```

could be that the detection of sqlite3 somehow doesn't work anymore?

--- a/configure.ac
+++ b/configure.ac
@@ -131,7 +131,7 @@ fi
 PKG_CHECK_MODULES(sqlite3, [sqlite3 >= 3.8.0], :, sqlite3_INTERNAL=yes)
 AM_CONDITIONAL(SQLITE3_INTERNAL, [test yes = "$sqlite3_INTERNAL"])
 if test yes = "$sqlite3_INTERNAL"; then
-   sqlite3_CFLAGS=-isystem '$(top_srcdir)/lib/sqlite'
+   sqlite3_CFLAGS='-isystem $(top_srcdir)/lib/sqlite'
    sqlite3_LIBS=
 fi
 AC_SUBST(sqlite3_CFLAGS)

